javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserClassDeclaration.java

```diff
@@ -238,10 +242,14 @@ public boolean canBeAssignedTo(ResolvedReferenceTypeDeclaration other) {
         if (this.getQualifiedName().equals(other.getQualifiedName())) {
             return true;
         }
-        ResolvedClassDeclaration superclass = (ResolvedClassDeclaration) getSuperClass().getTypeDeclaration();
-        if (superclass != null && superclass != this) {
-            if (superclass.canBeAssignedTo(other)) {
-                return true;
+
+        Optional<ResolvedReferenceType> optionalSuperClass = getSuperClass();
+        if(optionalSuperClass.isPresent()) {
```

Comment:
can't use `ifPresent()` here due to the return contained within it.. somebody later might want to edit it if you can see a way to do so, but this is low priority..